File: app/src/main/java/no/nav/tilbakemeldingsmottak/exceptions/EksterntKallException.java
package no.nav.tilbakemeldingsmottak.exceptions;

/**
 * Kastes dersom kall til opprettJournalpost eller opprettOppgave feiler.
 * Tjenesten skal håndtere dette og returnere 200 OK til sluttbruker.
 */
public class EksterntKallException extends AbstractGeneralException {

    public EksterntKallException(String message, Throwable cause) {
        super(message, cause);
    }

    public EksterntKallException(String message, Throwable cause, ErrorCode errorCode) {
        super(message, cause, errorCode);
    }
}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/exceptions/EksterntKallException.kt
package no.nav.tilbakemeldingsmottak.exceptions

/**
 * Kastes dersom kall til opprettJournalpost eller opprettOppgave feiler.
 * Tjenesten skal håndtere dette og returnere 200 OK til sluttbruker.
 */
open class EksterntKallException(
    override val message: String,
    override val cause: Throwable? = null,
    val errorCode: ErrorCode = ErrorCode.GENERAL_ERROR
) : RuntimeException(message)